src/app/auth/components/login/login.component.html

```diff
@@ -1,3 +1,15 @@
-<p>
-  login works!
-</p>
+<form #loginForm="ngForm" (ngSubmit)="onSubmit(loginForm)" novalidate>
+  <p>
+    <mat-form-field>
+      <input matInput placeholder="Username" name="username" ngModel required>
```

Comment:
it is not relevant for this course, as accessibility is not in focus for pilot, but just to mention (you probably already know this), in terms of screen readers, placeholder alone is not enough, we would need a label for each input.